### DISCUSSION TITLE
Simulator support of Vivado IPs

### DISCUSSION THREAD

**Discussion by KolosKoblasz** [cocotb/cocotb#3661] [Q&A] [answered]
Dear Community!

I would like to use CocoTb for simulating SystemVerilog modules which instantiate Vivado IPs.
What simulator do you suggest for this task?
It is a hobby project so I can't really spend thousands of dollars on licenses because this project won't generate any revenue.

Icarus / Verilator are awesome, but they can't simulate Vivado IPs.
VCS, / Questa and / Xcelium are unaffordable. (at least I suppose they are)
Vivado xsim is not supported by coco TB
Aldec Riviera Pro is deprecated according to this page: https://docs.cocotb.org/en/latest/simulator_support.html#synopsys-vcs
Aldec Active HDL is windows based, but I'm using Linux.

Thank you for your help in advance!
Kolos

**Comment by ktbarrett** [comment, marked as answer]
cocotb supports Riviera Pro. I think you are misreading the deprecation note. When running Riviera you have to use `make SIM=riviera`. `make SIM=aldec` was *also* supported as a synonym, but that was deprecated and removed.

I don't know the pricing on Riviera Pro, but I don't think it's cheap. I haven't tried, but the free versions of Modelsim may work for this task.

**Reply by KolosKoblasz** [reply]
Thank you very much!

**Comment by zaporozhets** [comment]
JFYI: https://github.com/themperek/cocotb-vivado
